extensions/tracing/src/main/java/io/opentelemetry/extension/tracing/ExtendedTracer.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.extension.tracing;
+
+import io.opentelemetry.api.trace.Span;
+import io.opentelemetry.api.trace.Tracer;
+import io.opentelemetry.context.Scope;
+import java.util.concurrent.Callable;
+
+/** Provides easy mechanisms for wrapping standard Java constructs with an OpenTelemetry Span. */
+@SuppressWarnings("InconsistentOverloads")
+public class ExtendedTracer {
+
+  private final Tracer tracer;
+
+  /** Create a new {@link ExtendedTracer} that wraps the provided Tracer. */
+  public ExtendedTracer(Tracer tracer) {
+    this.tracer = tracer;
+  }
+
+  /** Wrap the provided {@link Runnable} with a {@link Span} with the provided name. */
+  public Runnable wrap(String spanName, Runnable runnable) {
```

Comment:
Is `wrap` that useful? What sort of patterns is it good for? I sort of expect most people to want to `run` (similar to what `WithSpan` does when using the agent).

Reply:
It seems like it would be useful to use with frameworks that provide Runnables, and/or require them to be passed into other parts of the framework. Although, that also begs the question of how those frameworks might require context propagation work as well.

It's a fair question, for sure. I'm not sure how people might want to use these kinds of wrappers. If you think that just providing `run` and `call` methods would be less risky and/or more useful, happy to change to that for starters.

Reply:
simplified with just `run` and `call`. I think this makes sense as a first set of methods to provide.